File: cases/taint/scala/computed-property-positive/ComputedPropertyPositive.scala
package dataflowbench

object ComputedPropertyPositive {
  class Holder {
    var alpha: String = "clean"
    var beta: String = "clean"
  }

  def dfb_source(): String = { // DFB-SOURCE: computed-property-input
    "tainted"
  }

  def dfb_sink(value: String): Unit = {} // DFB-SINK: computed-property-sink

  def run(): Unit = {
    val holder = new Holder()
    val key = "alpha"
    val field = classOf[Holder].getDeclaredField(key)
    field.setAccessible(true)
    field.set(holder, dfb_source()) // DFB-WITNESS: computed-property-store
    dfb_sink(field.get(holder).asInstanceOf[String])
  }
}

File: cases/taint/php/computed-property-positive/computed_property_positive.php
<?php
class Holder
{
    public string $alpha = "clean";
    public string $beta = "clean";
}

function dfb_source(): string { // DFB-SOURCE: computed-property-input
    return "tainted";
}

function dfb_sink(string $value): void {} // DFB-SINK: computed-property-sink

function run(): void {
    $holder = new Holder();
    $key = "alpha";
    $holder->{$key} = dfb_source(); // DFB-WITNESS: computed-property-store
    dfb_sink($holder->{$key});
}
